src/OfflineTool.cpp

```diff
@@ -218,72 +218,85 @@ runCommand (const std::string& command,
-        doSerialize(input);
+    auto const serialize = [](
+        auto const& input, auto const&, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doSerialize(*input);
     };
-    auto const deserialize = [](auto const& input,
-            auto const&, auto const&, auto const&) {
-        doDeserialize(input);
+    auto const deserialize = [](
+        auto const& input, auto const&, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doDeserialize(*input);
     };
-    auto const sign = [](auto const& input,
-            auto const& keyFile, auto const&, auto const&) {
-        doSingleSign(input, keyFile);
+    auto const sign = [](
+        auto const& input, auto const& keyFile, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doSingleSign(*input, keyFile);
     };
-    auto const multi_sign = [](auto const& input,
-            auto const& keyFile, auto const&, auto const&) {
-        doMultiSign(input, keyFile);
+    auto const multi_sign = [](
+        auto const& input, auto const& keyFile, auto const&)
+    {
+        BOOST_ASSERT(input);
+        doMultiSign(*input, keyFile);
     };
-    auto const create_keyfile = [](auto const&,
-            auto const& keyFile, auto const& keyType, auto const& seed) {
+    auto const create_keyfile = [](
+        auto const& seed, auto const& keyFile, auto const& keyType)
+    {
         doCreateKeyfile(keyFile, keyType, seed);
     };
     static map<string, commandParams>
         const commandArgs = {
-            { "serialize", {0, 1, true, serialize } },
-            { "deserialize", {0, 1, true, deserialize } },
-            { "sign", {0, 1, true, sign } },
-            { "multi_sign", {0, 1, true, multi_sign } },
-            { "create_keyfile", {0, 0, false, create_keyfile } },
+            { "serialize", { "", serialize } },
+            { "deserialize", { "", deserialize } },
+            { "sign", { "", sign } },
+            { "multi_sign", { "", multi_sign } },
+            { "create_keyfile", { boost::none, create_keyfile } },
     };
 
     auto const iArgs = commandArgs.find(command);
 
     if (iArgs == commandArgs.end())
         throw std::runtime_error("Unknown command: " + command);
 
-    if (args.size() < iArgs->second.minParams ||
-            args.size() > iArgs->second.maxParams)
-        throw std::runtime_error("Syntax error: Wrong number of arguments");
-
-    auto const input = [&]
+    // getInputType has already resolved conflicts
+    boost::optional<std::string> input;
+    switch (inputType)
     {
-        if (args.size() == 0)
-            return iArgs->second.enableStdin ? getStdin() : string{};
-        else
-            return args[0];
-    }();
+    case InputType::readstdin:
+        input = boost::trim_copy(getStdin());
+        break;
+
+    case InputType::commandline:
+        if (args.size() != 1)
+            throw std::runtime_error("Syntax error: Wrong number of arguments");
```

Comment:
Nit: I think I'd rather see this error message than
`Unable to (de)serialize/sign ""` when neither `--stdin` nor a commandline argument is specified for signing or serialization commands.